docs/maven-plugin/src/main/java/org/keycloak/guides/maven/GuideBuilder.java

```diff
@@ -0,0 +1,57 @@
+package org.keycloak.guides.maven;
+
+import freemarker.template.TemplateException;
+import org.apache.maven.plugin.logging.Log;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.maven.project.MavenProject;
+
+public class GuideBuilder {
+
+    private final FreeMarker freeMarker;
+    private final File srcDir;
+    private final File targetDir;
+    private final Log log;
+
+    public GuideBuilder(File srcDir, File targetDir, Log log, MavenProject project) throws IOException {
+        this.srcDir = srcDir;
+        this.targetDir = targetDir;
+        this.log = log;
+
+        Map<String, Object> globalAttributes = new HashMap<>();
+        globalAttributes.put("ctx", new Context(srcDir));
+        globalAttributes.put("version", project.getVersion());
+        globalAttributes.put("properties", project.getProperties());
```

Comment:
```suggestion
```

This is not needed as you can get version from the `version` attribute, although that should probably be renamed to `client-version` to make it explicit and not risk ending up using the Keycloak server version when building docs.